modal/volume.py

```diff
@@ -276,6 +277,28 @@ async def reload(self):
         try:
             await self._do_reload()
         except GRPCError as exc:
+            # TODO(staffan): This is very brittle and janky. We should return the open files from the server.
+            if exc.message == "there are open files preventing the operation":
+                # Attempt to identify what open files are problematic and include the path in the error message.
+                # This is best-effort and not necessarily bulletproof, as the view of open files inside the container
+                # might differ from that outside - but it will at least catch common errors.
+                vol_path = f"/__modal/volumes/{self.object_id}"
+
+                cwd = PurePosixPath(os.readlink("/proc/self/cwd"))
+                if cwd.is_relative_to(vol_path):
+                    raise RuntimeError(f"{exc.message}: the current working directory is inside the volume")
+
+                open_paths = []
+                for fd in os.listdir("/proc/self/fd"):
```

Comment:
Doesn't work if subprocesses has open files, since this naively only looks at `/proc/self/fd`. Still better than nothing!

Should be simple enough to just traverse everything in `/proc/<pid>/fd` though, to catch those as well